### PR TITLE
Ingress extensions/v1beta1 -> networking.k8s.io/v1beta1

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
* first commit moves internal Ingress types, validation, and REST storage to the networking package
  * types and validation were copied with no changes (other than package fixup)
  * no externally visible API changes
* second commit copies the extensions/v1beta1 Ingress types to networking.k8s.io/v1beta1

**Which issue(s) this PR fixes**:
xref https://github.com/kubernetes/enhancements/issues/758
xref #43214

**Does this PR introduce a user-facing change?**:
```release-note
The Ingress API is now available via `networking.k8s.io/v1beta1`. `extensions/v1beta1` Ingress objects are deprecated and will no longer be served in v1.18.
```

/sig network
/cc @bowei @thockin

### CODE REVIEW

**Comment by liggitt** [comment]
fixed up lint and godep.json issues

**Comment by thockin** [comment]
Will try to get to this tomorrow.

File: pkg/apis/networking/types.go [hunk 1]
```diff
@@ -195,3 +195,172 @@ type NetworkPolicyList struct {
...
+	// Ingress, if left unspecified.
+	// +optional
+	Hosts []string
+	// SecretName is the name of the secret used to terminate SSL traffic on 443.
+	// Field is left optional to allow SSL routing based on SNI hostname alone.
+	// If the SNI host in a listener conflicts with the "Host" header field used
+	// by an IngressRule, the SNI host is used for termination and value of the
+	// Host header is used for routing.
+	// +optional
+	SecretName string
```
Comment (thockin) [inline pkg/apis/networking/types.go]: Because this is a cross-group, no-delta move, I won't harp, but we should probably make this a pointer before GA

**Comment by liggitt** [comment]
regenerated to pick up swagger doc deprecation warning from https://github.com/kubernetes/kubernetes/pull/73783, retagging

**Pull request merged**